cmd/pinniped/cmd/kubeconfig.go

```diff
@@ -268,7 +272,13 @@ func newExecConfig(deps kubeconfigDeps, flags getKubeconfigParams) (*clientcmdap
 
 	execConfig.InstallHint = flags.installHint
 	var err error
-	execConfig.Command, err = deps.getPathToSelf()
+	execConfig.Command, err = func() (string, error) {
+		if flags.commandExePath != "" {
+			return flags.commandExePath, nil
+		}
+		command, err := deps.getPathToSelf()
+		return filepath.Base(command), err
```

Comment:
Could we also strip off any `.exe` suffix? Maybe only do that if the current command is running on Windows?

If a user runs `C:\some\path\pinniped.exe get kubeconfig` on Windows, then we probably don't want the `.exe` suffix to be in the resulting kubeconfig by default, because that would not be needed on Windows and most likely would not work on other platforms.

Reply:
I chose not to do any filename manipulation because the user can choose to overwrite this if they don't like the default behavior.

I think in the end the persona who runs `pinniped get kubeconfig` has to provide their users with appropriate kubeconfigs. They could choose to generate `*nix` vs `Windows` kubeconfigs for distribution, if desired, by providing the appropriate input flags. I don't think we should guess for them how they want to use this.

Reply:
Fair enough. I've pushed this change.

Out of curiosity, I wonder if `pinniped get kubeconfig` should attempt any sort of validation on the generated kubeconfig. It's possible to run `C:\dir\not\in\path\pinniped.exe get kubeconfig` which would (now that I've pushed these changes) result in `command: pinniped` which technically would not be usable. But really it should be the user's responsibility to verify.

Reply:
That's a very interesting point!

Previously a user could download (or build) the CLI at any path and then use it to generate kubeconfigs and to perform logins with those kubeconfigs. In fact, I do that all the time when performing local testing. This PR will break that behavior, right? If I build the CLI locally, don't bother putting it on my path, and then run `./pinniped get kubeconfig` it is going to result (by default) in a kubeconfig that will not work for me or perhaps (even worse!) will use a different/older version of the CLI that it finds on my path. That's not great or usability.

That's an interesting tradeoff between making it easy for a user who builds or downloads the CLI somewhere temporarily (not on their path) versus for a user who runs `pinniped get kubeconfig` because they intend to distribute it widely (expects it to be on the path of their end users). Ideally we could make both use cases easy. Now you have me worried about this. :)

What could we do to avoid this being super confusing?
- Maybe `pinniped get kubeconfig` could notice this case and print a warning? Does golang offer a portable way to use the user's path to resolve the `command:` path to a binary? If so, `pinniped get kubeconfig` could check if the binary being used to run `pinniped get kubeconfig` and the binary that will be found at login time are identical binaries.
- Maybe it is a mistake to automatically remove the full path to the current binary by default? If the default behavior did not change, then the user who wants to distribute the kubeconfig widely could always use `--pinniped-cli-path=pinniped` and we could document that suggestion.

Reply:
It's possible to see if a binary is available on the path using https://pkg.go.dev/os/exec#LookPath. So I guess that behavior would look like this:

`C:\dir\not\in\path\pinniped.exe get kubeconfig` would resolve to `command: pinniped`, but the CLI would be able to determine that `pinniped` is not in the path by performing `exec.LookPath("pinniped")` and could print the warning (and `kubeconfig` generation should still succeed).

That behavior does make some sense to me but now it's just getting crazy. I wonder if we could allow users to request the full path of the binary being run by specifying something like `--pinniped-cli-path=-` or something like that.

Another option: We keep the existing behavior where the full path of the binary being run is put into the `kubeconfig`. If the caller wants the kubeconfig to be more generic they *must* specify `--pinniped-cli-path`.

Reply:
> now it's just getting crazy.

Yes my suggestion to do very smart warnings is quite complex. To detect that the user is going to accidentally run a different version of `pinniped` at login time, the CLI would probably have to do a binary diff or compare the sha sums of the two binaries (the one being used to run `pinniped get kubeconfig` vs. the one that will be found on the path). That's getting awfully complicated.

Perhaps it would be the least surprising for a user, and the least potentially confusing, if we did keep the current behavior where it embeds the full path to the binary used to run `pinniped get kubeconfig` like you suggested.

We could document that behavior more clearly in our docs and point out that a user can create a more portable kubeconfig by using `--pinniped-cli-path=pinniped` which will use whatever `pinniped` is found on the user's path at login time, and that this will also work for Windows users because golang on Windows will still find and run the binary from your path whether it is called `pinniped` or `pinniped.exe`. Perhaps we could add a new section to this doc: https://pinniped.dev/docs/howto/login/

Reply:
Let's keep the existing behavior when `--pinniped-cli-path` is not provided, which will cause the `pinniped` CLI to bake its own entire path into the kubeconfig that results from `pinniped get kubeconfig`.

If users want to overwrite this behavior, they can specify `--pinniped-cli-path` or modify the kubeconfig after generation. This PR should document usage of `--pinniped-cli-path` at https://pinniped.dev/docs/howto/login/.